src/main/java/com/dev/cinema/controller/OrderController.java

```diff
@@ -37,15 +38,26 @@ public OrderController(UserService userService,
     }
 
     @PostMapping("/complete")
-    public void completeOrder(@RequestParam Long userId) {
-        ShoppingCart shoppingCart = shoppingCartService
-                .getByUser(userService.get(userId));
-        orderService.completeOrder(shoppingCart);
+    public void completeOrder(Authentication authentication) {
+        Object principal = authentication.getPrincipal();
+        if (principal instanceof UserDetails) {
+            UserDetails userDetails = (UserDetails) principal;
+            ShoppingCart shoppingCart = shoppingCartService
+                    .getByUser(userService.findByEmail(userDetails.getUsername())
+                    .get());
+            orderService.completeOrder(shoppingCart);
+        }
     }
 
     @GetMapping
-    public List<OrderResponseDto> getOrdersHistory(@RequestParam Long userId) {
-        User user = userService.get(userId);
+    public List<OrderResponseDto> getOrdersHistory(Authentication authentication) {
+        Object principal = authentication.getPrincipal();
+        User user = null;
+        if (principal instanceof UserDetails) {
```

Comment:
let's assume in this case the principal can be safely casted to UserDetails, and remove this check

also, we get get email directly from Authentication
`authentication.getName()`

Reply:
done